api/ilos/tools/src/registerGracefulShutdown.ts

```diff
@@ -0,0 +1,40 @@
+export function registerGracefulShutdown(
+  // https://www.typescriptlang.org/docs/handbook/functions.html#this-parameters
+  this: any,
+  closeHandlers: Array<() => Promise<any>> = [],
+  timeout = 30,
+) {
+  // gracefully shutdown on SIGTERM or SIGINT signal
+  process.once('SIGTERM', gracefulShutDown.bind(this, 'SIGTERM'));
+  process.once('SIGINT', gracefulShutDown.bind(this, 'SIGINT'));
+
+  async function gracefulShutDown(signal = 'SIGTERM') {
+    console.info(`got kill signal (${signal}), starting graceful shut down`);
+
+    // shut down anyway after `timeout` seconds
+    if (timeout) {
+      setTimeout(() => {
+        console.error('could not finish in time, forcefully exiting');
+        process.exit(1);
+      }, timeout * 1000).unref();
```

Comment:
Pas sûr que `.unref()` soit utile ni conseillé.
https://www.bennadel.com/blog/3301-calling-timeout-unref-with-settimeout-does-not-appear-to-be-a-best-practice-in-node-js.htm